src/hooks/use-local-storage.ts

```diff
@@ -16,14 +13,14 @@ export function useLocalStorage<T>(
   key: string,
   initialValue: T
 ): [T, React.Dispatch<React.SetStateAction<T>>] {
-  const [storedValue, setStoredValue] = useState(
-    getItemFromLocalStorage(key) ?? initialValue
-  );
+  const [storedValue, setStoredValue] = useState(initialValue);
 
   useEffect(() => {
-    // Retrieve from localStorage
-    const item = getItemFromLocalStorage(key);
-    if (item) setStoredValue(item);
+    // initialize
+    if (typeof window !== "undefined") {
+      const stored = getItemFromLocalStorage(key);
+      setStoredValue(stored);
+    }
   }, [key]);
```

Comment:
We initialize the value of local storage on mount. This fixes the current server `ReferenceError: window is not defined` error but creates a flicker effect for the user if it varies from the default values.